### PR TITLE
APIMF 2881

### PR DESCRIPTION
**Pull request by tomsfernandez**

### CODE REVIEW

File: shared/src/main/scala/amf/client/plugins/AMFValidationPlugin.scala [hunk 1]
```diff
@@ -18,15 +20,12 @@ trait AMFValidationPlugin extends AMFPlugin with PlatformSecrets {
...
-      validations: EffectiveValidations,
-      platform: Platform,
-      env: Environment,
-      resolved: Boolean = false,
-      executionEnv: BaseExecutionEnvironment = platform.defaultExecutionEnvironment): Future[AMFValidationReport]
+  protected[amf] def getRemodValidatePlugins(): Seq[AMFValidatePlugin]
+}
+
+
+case class ValidationRequirement(builder: ValidationResolver, plugins: Seq[AMFValidatePlugin])
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/plugins/AMFValidationPlugin.scala]: Should be erased

File: shared/src/main/scala/amf/client/plugins/AMFValidationPlugin.scala [hunk 2]
```diff
@@ -18,15 +20,12 @@ trait AMFValidationPlugin extends AMFPlugin with PlatformSecrets {
...
+
+trait ValidationResolver {
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/plugins/AMFValidationPlugin.scala]: Should be erased

File: shared/src/main/scala/amf/client/remod/amfcore/plugins/validate/AMFValidatePlugin.scala [hunk 3]
```diff
@@ -1,9 +1,34 @@
...
-import amf.core.validation.AMFValidationReport
+import amf.core.unsafe.PlatformSecrets
+import amf.core.validation.{AMFValidationReport, EffectiveValidations}
+import amf.internal.environment.Environment
 
-private[remod] trait AMFValidatePlugin extends AMFPlugin[BaseUnit] {
-  def validate: AMFValidationReport
+import scala.concurrent.{ExecutionContext, Future}
+
+object ExpectedResolutionStatus {
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/amfcore/plugins/validate/AMFValidatePlugin.scala]: Should be erased

File: shared/src/main/scala/amf/client/remod/amfcore/plugins/validate/AMFValidatePlugin.scala [hunk 4]
```diff
@@ -1,9 +1,34 @@
...
+  val INDISTINCT: ExpectedResolutionStatus = IndistinctResolution()
+  val REQUIRED: ExpectedResolutionStatus = RequiredResolution()
+  val FORCE_REQUIRED: ExpectedResolutionStatus = ForceResolution()
+  val UNRESOLVED: ExpectedResolutionStatus = NoResolution()
+}
+
+trait ExpectedResolutionStatus
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/amfcore/plugins/validate/AMFValidatePlugin.scala]: Should be erased

**Review by looseale — DISMISSED** [review]

**Review by AgustinBettati — APPROVED** [review]

**Pull request merged**